src/main/java/com/codeit/otboo/domain/weather/service/WeatherServiceImpl.java

```diff
@@ -1,39 +1,66 @@
 package com.codeit.otboo.domain.weather.service;
 
-import com.codeit.otboo.domain.weather.component.KmaApiClient;
 import com.codeit.otboo.domain.weather.component.LocationConverter;
-import com.codeit.otboo.domain.weather.component.WeatherParser;
 import com.codeit.otboo.domain.weather.dto.WeatherDto;
+import com.codeit.otboo.domain.weather.entity.Weather;
 import com.codeit.otboo.domain.weather.entity.vo.LocationInfo;
+import com.codeit.otboo.domain.weather.repository.WeatherRepository;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import java.time.OffsetDateTime;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class WeatherServiceImpl implements WeatherService {
 
-    // Repository는 이제 사용하지 않습니다.
-    // private final WeatherRepository weatherRepository;
+    private final WeatherRepository weatherRepository;
     private final LocationConverter locationConverter;
-    private final KmaApiClient kmaApiClient;
-    private final WeatherParser weatherParser;
+    // 이제 실시간 API 호출이 필요 없으므로 KmaApiClient와 WeatherParser는 제거합니다.
+    // private final KmaApiClient kmaApiClient;
+    // private final WeatherParser weatherParser;
 
     @Override
+    @Transactional(readOnly = true) // 데이터 조회만 하므로 readOnly=true로 성능 최적화
     public List<WeatherDto> getWeather(double latitude, double longitude) {
         // 1. 위도/경도를 격자 X, Y로 변환합니다.
         LocationInfo locationInfo = locationConverter.toGrid(latitude, longitude);
 
-        // 2. 기상청 API를 실시간으로 호출하여 원본 JSON 데이터를 받습니다.
-        String rawJsonData = kmaApiClient.fetchWeatherForecast(locationInfo.x(), locationInfo.y());
+        // 2. Repository를 통해 DB에 저장된, 현재 시각 이후의 날씨 정보만 조회합니다.
+        List<Weather> weathers = weatherRepository.findWeathersByLocation(
+                locationInfo.x(),
+                locationInfo.y(),
+                OffsetDateTime.now() // 현재 시각
+        );
 
-        // 3. JSON 데이터를 파싱하고 가공하여 DTO 리스트로 변환합니다.
-        return weatherParser.parseAndGroup(rawJsonData, latitude, longitude);
+        // 3. 조회된 Entity 목록을 DTO 목록으로 변환하여 반환합니다.
+        return weathers.stream()
+                .map(this::mapToWeatherDto)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                
                return weathers.stream()
               .map(this::mapToWeatherDto)
               .toList();
```

불변 리스트를 반환하게 하서 방어적 코드를 구성하면 좋을거같아요

https://voidmelody.tistory.com/186